### PR TITLE
Remove workarounds for Microsoft My Day, Messenger, and Outlook

### PR DESCRIPTION
**Pull request by LumioseSil**
#### e5c038660e27bd020baf52f74a31b91dc467af5b
<pre>
Remove workarounds for Microsoft My Day, Messenger, and Outlook
<a href="https://bugs.webkit.org/show_bug.cgi?id=252816">https://bugs.webkit.org/show_bug.cgi?id=252816</a>

Reviewed by Brent Fulgham.

Microsoft My Day no longer exists as of 2008, so we can remove the
workarounds made for it. Same for Microsoft Messenger and Microsoft
Outlook as of 2010.

* Source/WebCore/WebCore.order: Removed mangled names of removed
  workarounds.

* Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:
  (isMicrosoftMessenger): Removed.
  (isMicrosoftMyDay): Removed.

* Source/WebCore/platform/RuntimeApplicationChecks.h:
  (isMicrosoftMessenger): Removed.
  (isMicrosoftMyDay): Removed.

* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj: Remove
  references to OutlookQuirksUserScript.js.

* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig: Ditto.

* Source/WebKitLegacy/mac/Misc/OutlookQuirksUserScript.js: Deleted.

* Source/WebKitLegacy/mac/Misc/WebKitVersionChecks.h: Removed
  WEBKIT_FIRST_VERSION_WITH_HTML5_PARSER.

* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
  (needsMicrosoftMessengerDOMDocumentWorkaround): Removed.

* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
  (defaultUsePreHTML5ParserQuirks): Remove calls to isMicrosoftMyDay.

* Source/WebKitLegacy/mac/WebView/WebView.mm:(needsOutlookQuirksScript):
  Removed.
  (_commonInitializationWithFrameName): Removed call to
  needsOutlookQuirksScript.
  (createOutlookQuirksUserScriptContents): Removed
  (_injectOutlookQuirksScript): Removed.

Canonical link: <a href="https://commits.webkit.org/262393@main">https://commits.webkit.org/262393@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/932e05d971649810f0152d760acf419320090d04

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1325 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1408 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2218 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1180 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1310 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1416 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1425 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1339 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1337 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1239 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1224 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2068 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1256 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1209 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1195 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1210 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1244 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2324 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1278 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1209 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1192 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1216 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/353 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1294 "Built successfully") | | | 
| | | | |

### CODE REVIEW

**Review by karlcow — APPROVED** [review]
* MSN Messenger became Windows Live Messenger and the [final release was on August 7, 2012](https://en.wikipedia.org/wiki/Windows_Live_Messenger). So yes safe to remove it. 
* [Microsoft To Do](https://todo.microsoft.com/tasks/) seems to includes a "My Day" feature, which is probably wildly different than the previous one. It was part of Entourage before that but was discontinued in 2008. Safe to remove.

**Comment by AtariDreams** [comment]
Can we merge these changes then?

**Comment by karlcow** [comment]
I let @brentfulgham validate the last step for C++ code side.

**Comment by AtariDreams** [comment]
Failures are unrelated to changes. However, I have to adjust the commit message, so there will be one more force-push.

**Comment by AtariDreams** [comment]
@brentfulgham Could we please merge this?

File: Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm [hunk 1]
```diff
@@ -106,29 +106,22 @@ bool defaultAllowContentSecurityPolicySourceStarToMatchAnyProtocol()
 
 bool defaultLoadDeferringEnabled()
 {
-    return !WebCore::MacApplication::isAdobeInstaller();
+    static bool isAdobeInstaller = WebCore::MacApplication::isAdobeInstaller();
+    return !isAdobeInstaller;
```
Comment (brentfulgham) [inline Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm]: This change is unnecessary. `WebCore::MacApplication::isAdobeInstaller` already caches its value.

File: Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm [hunk 2]
```diff
@@ -106,29 +106,22 @@ bool defaultAllowContentSecurityPolicySourceStarToMatchAnyProtocol()
...
 }
 
 bool defaultWindowFocusRestricted()
 {
-    return !WebCore::MacApplication::isHRBlock();
+    static bool isHRBlock = WebCore::MacApplication::isHRBlock();
+    return !isHRBlock;
```
Comment (brentfulgham) [inline Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm]: This change is unnecessary. `WebCore::MacApplication::isHRBlock` already caches its value.

File: Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm [hunk 3]
```diff
@@ -106,29 +106,22 @@ bool defaultAllowContentSecurityPolicySourceStarToMatchAnyProtocol()
...
 bool defaultUsePreHTML5ParserQuirks()
 {
-    // AOL Instant Messenger and Microsoft My Day contain markup incompatible
-    // with the new HTML5 parser. If these applications were linked against a
-    // version of WebKit prior to the introduction of the HTML5 parser, enable
-    // parser quirks to maintain compatibility. For details, see
-    // <https://bugs.webkit.org/show_bug.cgi?id=46134> and
-    // <https://bugs.webkit.org/show_bug.cgi?id=46334>.
-    static bool isApplicationNeedingParserQuirks = !WebKitLinkedOnOrAfter(WEBKIT_FIRST_VERSION_WITH_HTML5_PARSER)
-        && (WebCore::MacApplication::isMicrosoftMyDay());
```
Comment (brentfulgham) [inline Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:126]: Yeah, `WEBKIT_FIRST_VERSION_WITH_HTML5_PARSER` is from 2010 so this is good to remove.

Comment (brentfulgham) [reply Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:126]: In fact, that indicates that it should be safe to remove the `needsOutlookQuirksScript` Quirk, which would be a nice follow-up to this patch.

File: Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm [hunk 4]
```diff
@@ -106,29 +106,22 @@ bool defaultAllowContentSecurityPolicySourceStarToMatchAnyProtocol()
...
-
     // Mail.app must continue to display HTML email that contains quirky markup.
     static bool isAppleMail = WebCore::MacApplication::isAppleMail();
 
-    return isApplicationNeedingParserQuirks || isAppleMail;
+    return isAppleMail;
```
Comment (brentfulgham) [inline Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm]: This should just become:

```
return webcore::MacApplication::isAppleMail();
```

**Review by brentfulgham — CHANGES_REQUESTED** [review]
I think this looks good, but I'd like you to remove the unneeded changes in `defaultLoadDeferringEnabled` and `defaultWindowFocusRestricted`, and revise the return value for `defaultUsePreHTML5ParserQuirks` as I suggest in the detailed notes.

Once those changes are in place I think this is ready to land.

**Comment by AtariDreams** [comment]
Done! @brentfulgham

**Review by brentfulgham — APPROVED** [review]
Looks good!

**Comment by AtariDreams** [comment]
@brentfulgham Can we please merge this?

**Comment by brentfulgham** [comment]
> @AtariDreams - do you want me to add 'merge-queue' label?

We can do that once the bots have finished processing.

**Comment by Ahmad-S792** [comment]
> > @AtariDreams - do you want me to add 'merge-queue' label?
> 
> We can do that once the bots have finished processing.

Yes - I was asking just to confirm that there are no more commits coming. :-)

**Comment by AtariDreams** [comment]
> > > @AtariDreams - do you want me to add 'merge-queue' label?
> > 
> > 
> > We can do that once the bots have finished processing.
> 
> Yes - I was asking just to confirm that there are no more commits coming. :-)

No more commits

**Comment by Ahmad-S792** [comment]
@brentfulgham - I am marking this commit as 'merge-queue' after checking on results.webkit.org for failing `mac-AS-debug-WK2` test `fast/text/web-font-load-invisible-during-loading.html` and it seems that it failed on other commits as well and suggesting it to be flaky failure.

**Pull request merged**